src/protocol/ipa/mod.rs

```diff
@@ -534,20 +534,11 @@ pub mod tests {
                 .await
                 .unwrap()
             })
-            .await
-            .reconstruct();
+            .await;
 
-        assert_eq!(EXPECTED.len(), result.len());
-
-        for (i, expected) in EXPECTED.iter().enumerate() {
-            assert_eq!(
-                *expected,
-                [
-                    result[i].breakdown_key.as_u128(),
-                    result[i].trigger_value.as_u128()
-                ]
-            );
-        }
+        assert_eq!(EXPECTED.len(), result0.len());
+        assert_eq!(EXPECTED.len(), result1.len());
+        assert_eq!(EXPECTED.len(), result2.len());
```

Comment:
Wait a minute... we lost the check that the output was correct?

Reply:
Let me bring it back with the validator. I was not sure which way to go .

Reply:
Yeah, I think calling the validator and then checking the results would be better.